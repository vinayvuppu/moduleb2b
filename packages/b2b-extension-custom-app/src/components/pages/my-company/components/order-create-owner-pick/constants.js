// eslint-disable-next-line
export const OWNER_TYPES = {
  OWNER_ME: 'me',
  OWNER_EMPLOYEE: 'employee',
  OWNER_COMPANY: 'company',
};
