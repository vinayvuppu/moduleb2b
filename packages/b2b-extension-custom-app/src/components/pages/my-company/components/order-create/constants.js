// Tab names for the steps in the header which should be defined
// in order because they determine the animation direction when
// switching between tabs
// eslint-disable-next-line import/prefer-default-export
export const ORDER_CREATE_TAB_NAMES = {
  ADD_LINE_ITEMS: 'lineitems',
  SET_OWNER: 'owner',
  SET_SHIPPING_METHOD: 'shipping-method',
  CONFIRMATION: 'confirmation',
};

export const ORDER_STATES = {
  OPEN: 'open',
  PENDING_APPROVAL: 'pendingApproval',
};
