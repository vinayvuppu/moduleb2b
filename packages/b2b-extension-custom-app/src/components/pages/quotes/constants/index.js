// eslint-disable-next-line
export const QUOTE_TYPES = {
  INITIAL: 'initial',
  SUBMITTED: 'submitted',
  APPROVED: 'approved',
  CLOSED: 'closed',
  DECLINED: 'declined',
  EXPIRED: 'expired',
  PLACED: 'placed',
};

export const QUOTE_DETAILS_TAB_NAMES = {
  GENERAL: 'general',
  COMMENTS: 'comments',
};
