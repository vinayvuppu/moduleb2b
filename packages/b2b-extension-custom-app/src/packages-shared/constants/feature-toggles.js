// A dependant flag configured in LaunchDarkly, which
// depends on the state of the PIM_SEARCH flag, where it
// serves the opposite of the state of the PIM_SEARCH flag
// e.g. if PIM_SEARCH is `true`, LEGACY_PRODUCT_LIST will serve `false` and vice-versa
export const LEGACY_PRODUCT_LIST = 'legacyProductList';
export const CAN_VIEW_DASHBOARD = 'canViewDashboard';

// Defaulted through the `allFeatureToggles` query in the mc-http-proxy
export const PIM_SEARCH = 'pimSearch';
export const CUSTOM_APPLICATIONS = 'customApplications';

export default {
  [CAN_VIEW_DASHBOARD]: true,
  [LEGACY_PRODUCT_LIST]: true,
};
