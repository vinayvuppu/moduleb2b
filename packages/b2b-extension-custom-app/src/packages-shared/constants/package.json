{
  "name": "@commercetools-local/constants",
  "version": "1.0.0",
  "description": "",
  "license": "Proprietary",
  "private": true,
  "dependencies": {},
  "devDependencies": {}
}
