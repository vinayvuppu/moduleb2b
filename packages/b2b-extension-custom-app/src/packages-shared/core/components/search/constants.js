const SEARCH_FILTERS_EVENTS = {
  updateFilterValue: 'updateFilterValue',
  toggleEditMode: 'toggleEditMode',
  cancelFilterChanges: 'cancelFilterChanges',
  removeFilterFromField: 'removeFilterFromField',
  clearAllFilters: 'clearAllFilters',
  addFilterToField: 'addFilterToField',
  clearFiltersFromField: 'clearFiltersFromField',
  addField: 'addField',
  applyFiltersAndRemoveEmpty: 'applyFiltersAndRemoveEmpty',
  applyFilters: 'applyFilters',
  removeFilterTagFromField: 'removeFilterTagFromField',
  clearAll: 'clearAll',
};
export default SEARCH_FILTERS_EVENTS;
