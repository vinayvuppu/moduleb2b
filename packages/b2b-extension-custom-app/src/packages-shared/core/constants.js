export const PRODUCT_ACTIONS = {
  PUBLISH: 'publish',
  UNPUBLISH: 'unpublish',
};

export const PRODUCT_STATUSES = {
  PUBLISHED: 'published',
  UNPUBLISHED: 'unpublished',
  MODIFIED: 'modified',
};

export const FILTER_TYPES = {
  lessThan: 'lessThan',
  moreThan: 'moreThan',
  equalTo: 'equalTo',
  range: 'range',
  missing: 'missing',
  missingIn: 'missingIn',
  in: 'in',
};

export const INITIALIZE_SEARCH_SLICE = 'INITIALIZE_SEARCH_SLICE';
export const SET_SEARCH_FILTER_STATE = 'SET_SEARCH_FILTER_STATE';
export const SET_SEARCH_FILTER_RESULT = 'SET_SEARCH_FILTER_RESULT';
export const REMOVE_ENTITY_BY_ID_FROM_SEARCH_RESULTS =
  'REMOVE_ENTITY_BY_ID_FROM_SEARCH_RESULTS';

export const FIELD_TYPES = {
  Money: 'Money',
  LocalizedString: 'LocalizedString',
  Time: 'Time',
  DateTime: 'DateTime',
  Boolean: 'Boolean',
  String: 'String',
  Enum: 'Enum',
  LocalizedEnum: 'LocalizedEnum',
  Number: 'Number',
  Date: 'Date',
  Reference: 'Reference',
  Set: 'Set',
};

export const permissions = {
  ViewCustomerGroups: 'ViewCustomerGroups',
  ViewStates: 'ViewStates',
};

export const QUERY_MAX_LIMIT = 500;
// PRODUCT_PROJECTION
export const PRODUCT_SET_DRAFT = 'PRODUCT_SET_DRAFT';
export const MODIFIED_PRODUCTS_FETCHED = 'MODIFIED_PRODUCTS_FETCHED';
export const PRODUCTS_FETCHED = 'PRODUCTS_FETCHED';
export const PRODUCT_FETCHED = 'PRODUCT_FETCHED';
export const PRODUCT_UPDATED_WITH_IMAGE_UPLOAD =
  'PRODUCT_UPDATED_WITH_IMAGE_UPLOAD';
export const PRODUCT_IMAGE_UPLOADED = 'PRODUCT_IMAGE_UPLOADED';
export const PRODUCT_SET_CHECK_SELECTION = 'PRODUCT_SET_CHECK_SELECTION';
export const PRODUCT_SET_ALL_CHECK_SELECTION =
  'PRODUCT_SET_ALL_CHECK_SELECTION';
export const PRODUCT_UPDATED = 'PRODUCT_UPDATED';
export const PRODUCT_CREATED = 'PRODUCT_CREATED';
export const SET_VISIBLE_PRODUCT = 'SET_VISIBLE_PRODUCT';
export const PRODUCT_STATE_TRANSITIONED = 'PRODUCT_STATE_TRANSITIONED';
export const PRODUCTS_DELETED_IN_LIST = 'PRODUCTS_DELETED_IN_LIST';
export const PRODUCT_MASTER_VARIANT_CHANGED = 'PRODUCT_MASTER_VARIANT_CHANGED';
// PRODUCT_TYPE_SETTINGS
export const PRODUCT_TYPE_SETTINGS_FETCHED = 'PRODUCT_TYPE_SETTINGS_FETCHED';
export const PRODUCT_TYPE_SETTINGS_CHANGED = 'PRODUCT_TYPE_SETTINGS_CHANGED';
export const PRODUCT_TYPE_SETTINGS_DELETED = 'PRODUCT_TYPE_SETTINGS_DELETED';
// PRODUCT VIEW
export const SET_VIEW_MODE = 'SET_VIEW_MODE';
export const VIEW_MODES = {
  PRODUCTS_LIST: 'PRODUCTS_LIST',
  BULK_UPDATE: 'BULK_UPDATE',
};

// PRODUCT_SETTINGS
export const ADD_PRODUCT_LIST_VIEW_SETTINGS_TO_LIST =
  'ADD_PRODUCT_LIST_VIEW_SETTINGS_TO_LIST';
export const REMOVE_PRODUCT_LIST_VIEW_SETTINGS_FROM_LIST =
  'REMOVE_PRODUCT_LIST_VIEW_SETTINGS_FROM_LIST';
export const SET_PRODUCT_LIST_VIEW_SETTINGS_LIST =
  'SET_PRODUCT_LIST_VIEW_SETTINGS_LIST';
export const SET_ACTIVE_PRODUCT_LIST_VIEW_SETTINGS =
  'SET_ACTIVE_PRODUCT_LIST_VIEW_SETTINGS';

// ATTRIBUTE CONSTRAINTS

export const ATTRIBUTE_CONSTRAINTS = {
  SAME_FOR_ALL: 'SameForAll',
  NONE: 'None',
  UNIQUE: 'Unique',
  COMBINATION_UNIQUE: 'CombinationUnique',
};

// STATES
export const INITIAL_PRODUCT_STATES_FETCHED = 'INITIAL_PRODUCT_STATES_FETCHED';
export const PRODUCT_STATES_FETCHED = 'PRODUCT_STATES_FETCHED';
// PRODUCT SEARCH AND FILTER
export const PRODUCT_SEARCH_TEXT = 'filter_text';
export const PRODUCT_SEARCH_SORTING = 'list_sorting';
export const PRODUCT_SEARCH_PAGE = 'list_page';
export const PRODUCT_SEARCH_FILTER_PRODUCT_TYPE = 'filter_productTypes';
export const PRODUCT_SEARCH_FILTER_ATTRIBUTES = 'filter_customAttributes';
export const PRODUCT_SEARCH_FILTER_CATEGORIES = 'filter_categories';
export const PRODUCT_SEARCH_FILTER_STATUS = 'filter_status';
export const PRODUCT_SEARCH_FILTER_STATES = 'filter_states';
export const PRODUCT_SEARCH_FILTER_MISSING = 'filter_missing';
export const PRODUCT_SEARCH_FILTERS_VISIBLE = 'filter_tags_visible';
export const PRODUCT_SEARCH_FILTERS_EXCLUDE_SUBCATEGORIES =
  'filter_exclude_subcategories';
// PRODUCT_TYPES
export const PRODUCT_TYPES_FETCHED = 'PRODUCT_TYPES_FETCHED';
export const PRODUCT_TYPE_FETCHED = 'PRODUCT_TYPE_FETCHED';
export const PRODUCT_TYPE_CREATED = 'PRODUCT_TYPE_CREATED';
export const PRODUCT_TYPE_DELETED = 'PRODUCT_TYPE_DELETED';
export const PRODUCT_TYPES_FOR_FILTER_FETCHED =
  'PRODUCT_TYPES_FOR_FILTER_FETCHED';
export const PRODUCT_TYPES_FOR_ATTRIBUTES_FETCHED =
  'PRODUCT_TYPES_FOR_ATTRIBUTES_FETCHED';
// RECOMMENDED_CATEGORIES
export const RECOMMENDED_CATEGORIES_FETCHED = 'RECOMMENDED_CATEGORIES_FETCHED';
export const RECOMMENDED_CATEGORIES_STATUS = 'RECOMMENDED_CATEGORIES_STATUS';
export const RECOMMENDED_CATEGORIES_SEARCH_PROPERTIES = {
  Attribute: 'Attribute',
  MachineLearning: 'MachineLearning',
  ProductType: 'ProductType',
};
export const RECOMMENDED_CATEGORIES_STATUS_TYPE = {
  ON: 'on',
  OFF: 'off',
  PROCESSING: 'processing',
};
// PRODUCT BULK UPDATES
export const START_BULK_UPDATE = 'BULK_START_BULK_UPDATE';
export const ADD_BULK_UPDATE_SUCCESS = 'BULK_ADD_BULK_UPDATE_SUCCESS';
export const ADD_BULK_UPDATE_FAILURE = 'BULK_ADD_BULK_UPDATE_FAILURE';
export const BULK_NEXT_STEP = 'BULK_NEXT_STEP';
export const BULK_SET_SELECTED_ATTRIBUTES = 'BULK_SET_SELECTED_ATTRIBUTES';
export const BULK_TOGGLE_CONFIRMATION_DIALOG =
  'BULK_TOGGLE_CONFIRMATION_DIALOG';
export const BULK_LEAVE_BULK_UPDATE = 'BULK_LEAVE_BULK_UPDATE';
