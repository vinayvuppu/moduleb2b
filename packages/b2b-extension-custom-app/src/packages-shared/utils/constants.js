export const PRODUCT_STATUSES = {
  PUBLISHED: 'published',
  UNPUBLISHED: 'unpublished',
  MODIFIED: 'modified',
};

export const FILTER_TYPES = {
  lessThan: 'lessThan',
  moreThan: 'moreThan',
  equalTo: 'equalTo',
  range: 'range',
  missing: 'missing',
  missingIn: 'missingIn',
  in: 'in',
};

export const FIELD_TYPES = {
  Money: 'Money',
  LocalizedString: 'LocalizedString',
  Time: 'Time',
  DateTime: 'DateTime',
  Boolean: 'Boolean',
  String: 'String',
  Enum: 'Enum',
  LocalizedEnum: 'LocalizedEnum',
  Number: 'Number',
  Date: 'Date',
  Reference: 'Reference',
  Set: 'Set',
};

export const PRECISION_TYPES = {
  highPrecision: 'highPrecision',
  centPrecision: 'centPrecision',
};

export const DefaultPageSizes = [20, 50, 100];
export const DefaultPageSize = DefaultPageSizes[0];
