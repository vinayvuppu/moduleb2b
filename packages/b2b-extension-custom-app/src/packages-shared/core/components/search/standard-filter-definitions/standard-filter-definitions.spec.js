import { FILTER_TYPES } from '../../../constants';
import {
  filterTypeDateMessages,
  filterTypeNumberMessages,
  filterTypeMissingMessages,
  createDateDefinitionsMap,
  createNumberDefinitionsMap,
  createMissingDefinitionsMap,
  createMissingInDefinitionsMap,
  createReferenceDefinitionsMap,
  createTypesDefinitionsMap,
  createOptionsDefinitionsMap,
  createCreatableOptionsDefinitionsMap,
  createCustomFieldDefinitionsMap,
} from './standard-filter-definitions';

const intl = { formatMessage: message => message.id };

describe('constants', () => {
  it('export filter type keys', () => {
    expect(FILTER_TYPES).toMatchSnapshot();
  });
});

describe('messages', () => {
  it('export filter types date messages', () => {
    expect(filterTypeDateMessages).toBeDefined();
  });
  it('export filter types number messages', () => {
    expect(filterTypeNumberMessages).toBeDefined();
  });
  it('export filter types missing messages', () => {
    expect(filterTypeMissingMessages).toBeDefined();
  });
});

describe('definitions', () => {
  describe('createDateDefinitionsMap', () => {
    it('should return a standard config for a "date" field', () => {
      expect(createDateDefinitionsMap(intl)).toMatchSnapshot();
    });
  });

  describe('createNumberDefinitionsMap', () => {
    it('should return a standard config for a "numver" field', () => {
      expect(
        createNumberDefinitionsMap(intl, { numberFormat: 'en' })
      ).toMatchSnapshot();
    });
  });

  describe('createMissingDefinitionsMap', () => {
    it('should return a standard config for a "missing" field', () => {
      expect(createMissingDefinitionsMap(intl)).toMatchSnapshot();
    });
  });

  describe('createMissingInDefinitionsMap', () => {
    it('should return a standard config for a "missingIn" field', () => {
      expect(
        createMissingInDefinitionsMap(intl, {
          options: [{ label: 'en', value: 'en' }],
        })
      ).toMatchSnapshot();
    });
  });

  describe('createReferenceDefinitionsMap', () => {
    it('should return a standard config for a "reference" field', () => {
      expect(
        createReferenceDefinitionsMap(intl, {
          loadItems: jest.fn(),
          mapItemToOption: jest.fn(),
          type: 'category',
        })
      ).toMatchSnapshot();
    });
  });

  describe('createTypesDefinitionsMap', () => {
    it('should return a standard config for a "types" field', () => {
      expect(
        createTypesDefinitionsMap(intl, {
          options: [{ label: 'en', value: 'en' }],
        })
      ).toMatchSnapshot();
    });
  });

  describe('createOptionsDefinitionsMap', () => {
    it('should return a standard config for a "options" field', () => {
      expect(
        createOptionsDefinitionsMap(intl, {
          options: [
            { label: 'en', value: { key: 'value-key', label: 'value-label' } },
          ],
        })
      ).toMatchSnapshot();
    });
  });

  describe('createCreatableOptionsDefinitionsMap', () => {
    it('should return a standard config for a "creatable options" field', () => {
      expect(createCreatableOptionsDefinitionsMap(intl, {})).toMatchSnapshot();
    });
  });

  describe('createCustomFieldDefinitionsMap', () => {
    it('should return a config for a "custom field" filter', () => {
      expect(createCustomFieldDefinitionsMap(intl, {})).toMatchSnapshot();
    });
  });
});
