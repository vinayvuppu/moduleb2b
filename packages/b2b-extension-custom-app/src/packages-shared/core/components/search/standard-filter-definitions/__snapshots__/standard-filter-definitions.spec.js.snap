// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants export filter type keys 1`] = `
Object {
  "equalTo": "equalTo",
  "in": "in",
  "lessThan": "lessThan",
  "missing": "missing",
  "missingIn": "missingIn",
  "moreThan": "moreThan",
  "range": "range",
}
`;

exports[`definitions createCreatableOptionsDefinitionsMap should return a standard config for a "creatable options" field 1`] = `
Object {
  "in": Object {
    "canBeAppliedMultipleTimes": true,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.Text.in",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;

exports[`definitions createCustomFieldDefinitionsMap should return a config for a "custom field" filter 1`] = `
Object {
  "customField": Object {
    "canBeAppliedMultipleTimes": true,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;

exports[`definitions createDateDefinitionsMap should return a standard config for a "date" field 1`] = `
Object {
  "equalTo": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Date.equalTo",
    "tagComponent": [Function],
  },
  "lessThan": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Date.lessThan",
    "tagComponent": [Function],
  },
  "moreThan": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Date.moreThan",
    "tagComponent": [Function],
  },
  "range": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Date.range",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;

exports[`definitions createMissingDefinitionsMap should return a standard config for a "missing" field 1`] = `
Object {
  "missing": Object {
    "canBeAppliedMultipleTimes": false,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.missing",
    "tagComponent": [Function],
  },
}
`;

exports[`definitions createMissingInDefinitionsMap should return a standard config for a "missingIn" field 1`] = `
Object {
  "missingIn": Object {
    "canBeAppliedMultipleTimes": false,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.missingIn",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;

exports[`definitions createNumberDefinitionsMap should return a standard config for a "numver" field 1`] = `
Object {
  "equalTo": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Number.equalTo",
    "tagComponent": [Function],
    "validator": [Function],
  },
  "lessThan": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Number.lessThan",
    "tagComponent": [Function],
    "validator": [Function],
  },
  "moreThan": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Number.moreThan",
    "tagComponent": [Function],
    "validator": [Function],
  },
  "range": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Number.range",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;

exports[`definitions createOptionsDefinitionsMap should return a standard config for a "options" field 1`] = `
Object {
  "equalTo": Object {
    "canBeAppliedMultipleTimes": false,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.Text.equalTo",
    "tagComponent": [Function],
    "validator": [Function],
  },
  "in": Object {
    "canBeAppliedMultipleTimes": false,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.Text.in",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;

exports[`definitions createReferenceDefinitionsMap should return a standard config for a "reference" field 1`] = `
Object {
  "equalTo": Object {
    "filterComponent": [Function],
    "label": "Search.Filters.Definitions.Reference.equalTo",
    "tagComponent": [Function],
  },
}
`;

exports[`definitions createTypesDefinitionsMap should return a standard config for a "types" field 1`] = `
Object {
  "equalTo": Object {
    "canBeAppliedMultipleTimes": false,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.Text.equalTo",
    "tagComponent": [Function],
    "validator": [Function],
  },
  "in": Object {
    "canBeAppliedMultipleTimes": false,
    "filterComponent": [Function],
    "isEmptyFilter": [Function],
    "label": "Search.Filters.Definitions.Text.in",
    "tagComponent": [Function],
    "validator": [Function],
  },
}
`;
